src/auth/http-poller-with-promises.ts

```diff
@@ -0,0 +1,51 @@
+import to from 'await-to-js';
+
+import createHttpPoller, { HttpPoller } from './http-poller';
+
+export type RetryingPollerProps = {
+  pollFunction: () => Promise<Response | undefined>;
+  delayInMs?: number;
+  maxRetries?: number;
+};
+
+export default async function retryPollingUntilSuccessful({
```

Comment:
To me this seems like a module that on the general level tries something again and again until successful. In this case it's tightly coupled to the HttpPoller interface, but it doesn't need to be. The general case has already been implemented, for example in [retry](https://www.npmjs.com/package/retry) or [p-retry](https://www.npmjs.com/package/p-retry). Could those be used instead of reinventing the wheel?

Reply:
The idea is to couple this to the http-poller as it does what is needed. After adding the onSuccess -callback, of course. There wasn't much to reinvent as we already have a module that makes async polling. 

Adding a new module seemed like an overkill vs few new lines of code + tests.

Reply:
OK. That's fine.